tests/models/vitpose/test_modeling_vitpose.py

```diff
@@ -0,0 +1,259 @@
+        self.config_tester.check_config_arguments_init()
+
+    @unittest.skip(reason="ViTPose does not use inputs_embeds")
+    def test_inputs_embeds(self):
+        pass
+
+    @unittest.skip(reason="ViTPose does not support training yet")
+    def test_training(self):
+        pass
+
+    @unittest.skip(reason="ViTPose does not support training yet")
+    def test_training_gradient_checkpointing(self):
+        pass
+
+    @unittest.skip(reason="ViTPose does not support training yet")
+    def test_training_gradient_checkpointing_use_reentrant(self):
+        pass
+
+    @unittest.skip(reason="ViTPose does not support training yet")
+    def test_training_gradient_checkpointing_use_reentrant_false(self):
+        pass
+
+    def test_model_common_attributes(self):
+        config, _ = self.model_tester.prepare_config_and_inputs_for_common()
+
+        for model_class in self.all_model_classes:
+            model = model_class(config)
+            self.assertIsInstance(model.get_input_embeddings(), (nn.Module))
+            x = model.get_output_embeddings()
+            self.assertTrue(x is None or isinstance(x, nn.Linear))
+
+    def test_forward_signature(self):
+        config, _ = self.model_tester.prepare_config_and_inputs_for_common()
+
+        for model_class in self.all_model_classes:
+            model = model_class(config)
+            signature = inspect.signature(model.forward)
+            # signature.parameters is an OrderedDict => so arg_names order is deterministic
+            arg_names = [*signature.parameters.keys()]
+
+            expected_arg_names = ["pixel_values"]
+            self.assertListEqual(arg_names[:1], expected_arg_names)
+
+    def test_for_pose_estimation(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_pose_estimation(*config_and_inputs)
+
+    @slow
+    def test_model_from_pretrained(self):
+        # TODO update organization
+        model_name = "nielsr/vitpose-base-simple"
+        model = ViTPoseForPoseEstimation.from_pretrained(model_name)
+        self.assertIsNotNone(model)
+
+
+# We will verify our results on an image of cute cats
+def prepare_img():
+    image = Image.open("./tests/fixtures/tests_samples/COCO/000000039769.png")
+    return image
+
+
+@require_torch
+@require_vision
+class ViTPoseModelIntegrationTest(unittest.TestCase):
+    @cached_property
+    def default_image_processor(self):
+        # TODO update organization
+        return ViTPoseImageProcessor.from_pretrained("nielsr/vitpose-base-simple") if is_vision_available() else None
+
+    @slow
+    def test_inference(self):
+        image_processor = self.default_image_processor
+        # TODO update organization
+        model = ViTPoseForPoseEstimation.from_pretrained("nielsr/vitpose-base-simple")
+
+        image = prepare_img()
+        boxes = [[[412.8, 157.61, 53.05, 138.01], [384.43, 172.21, 15.12, 35.74]]]
```

Comment:
Could you make this batched?

Reply:
The reason this takes a list of bounding boxes rather than an array is because the number of bounding boxes might differ per image.

This is similar to how `DetrImageProcessor` for instance accepts the [annotations flag](https://github.com/huggingface/transformers/blob/96eb06286b63c9c93334d507e632c175d6ba8b28/src/transformers/models/detr/image_processing_detr.py#L1231) which supports variable number of boxes per image.

Reply:
@NielsRogge @amyeroberts Should we consider to accept batched input rather than list here?

https://github.com/huggingface/transformers/blob/3d1824d9b474ebd3c35abe564472290a1ddd87ff/src/transformers/models/vitpose/image_processing_vitpose.py#L373

Reply:
I think we should be able to accept both. If the number of bboxes varies, then we'll have to accept nested lists, but it should also be possible to pass in an array too

Reply:
I have fixed to accept np.array in this method also. However, the testing value is remained as present, because I think the usage case of nested list should be more frequent!

https://github.com/SangbumChoi/transformers/commit/a55a95564c9d84770fb3dcffe2bc2a1f9f5c6a8b